### ISSUE TITLE
Add Audio Fade Handles Shortcut

### ISSUE THREAD

**Issue by randomeizer** [fcpxhacks/fcpxhacks#3]
Note: this has already been implemented in FCPX 10.3.

**Comment by latenitefilms** [comment]
@randomeizer - The new "Apply Audio Fades" shortcut in FCPX 10.3 is pretty annoying in my opinion. Rather than applying a default fade length, I'd much rather it apply a fade in/out to wherever the playhead is located, so this is something I still definitely want to fix up.

**Comment by randomeizer** [comment]
Ah, ok. Do you see it as being two separate operations - i.e. 'Fade In to Playhead' and 'Fade Out from Playhead'?

**Comment by latenitefilms** [comment]
Yes, I think this is probably the best way to do this (same as other NLE's such as Avid Media Composer) - although it could be interesting to see if there's a way to "detect" if it should be applying the fade as an "in" or an "out" based on the distance between the start/end and the playhead.

This is a feature I've been wanting to add to FCPX Hacks since the beginning - but it's really tricky, as the only way to manipulate the fade handles is via the mouse.

**Comment by latenitefilms** [comment]
@randomeizer - This is something I would absolutely LOVE for the job I'm currently working on.

I want to just be able to select two clips, press a shortcut, and it will automatically to an audio fade handle cross-fade for me (see attached screenshot).

I know it's been previously discussed on the Facebook group about do this using a series of keyboard shortcuts in sequence - but it proved a bit unreliable, so I think the best way to do it is via Clipboard manipulation.

Obviously we've still got a fair amount to do to really solve clipboard manipulation properly - but I wonder if it's something we could "bash out" more quickly, to just get this feature working?

The re-archiving of the Pasteboard XML is obviously a problem - but I wonder if FCPX has the ability to handle things, if you pass it along un-archived plist data? In theory the data should be the same - it's just organised in a different manner, so FCPX MAY be ok with it in any format (as I assume they're just using the macOS Pasteboard Frameworks for handling anyway?

I might try and have a play in between editing - but if you have a chance to play, that would be amazing. Sorry - I know you've got a million things to do both outside of CommandPost and fixing up the existing branches - so no stress if you simply don't have time.

Adventure, adventure!

<img width="302" alt="screen shot 2017-08-12 at 2 57 50 pm" src="https://user-images.githubusercontent.com/22286696/29238025-abc8670c-7f6e-11e7-908d-74afca3af065.png">

**Comment by latenitefilms** [comment]
Would be great to have a "delete and add cross fade" shortcut too, for when you're quickly trying to join bits of an interview together.

**Comment by latenitefilms** [comment]
As an interim solution, we could try just do something like this?

https://www.youtube.com/watch?v=ypoVG3vZDLU

**Comment by latenitefilms** [comment]
Note to self, you can change the audio fade duration using:

```
defaults write com.apple.FinalCut FFDefaultFadeAudioDuration -float 2
```
